src/main/java/com/google/devtools/build/lib/remote/disk/DiskAndRemoteCacheClient.java

```diff
@@ -75,7 +76,7 @@ public void close() {
   public ListenableFuture<Void> uploadFile(
       RemoteActionExecutionContext context, Digest digest, Path file) {
     ListenableFuture<Void> future = diskCache.uploadFile(context, digest, file);
-    if (shouldUploadLocalResultsToRemoteCache(options, context.getSpawn())) {
+    if (shouldUploadContentToRemoteCache(options, context.getSpawn())) {
```

Comment:
Can we changed to `options.isRemoteExecutionEnabled() || shouldUploadLocalResultsToRemoteCache()`?
  1. The "content" in `shouldUploadContent...` is not clear.
  2. To reflect that we check `options.isRemoteExecutionEnabled()` inside `findMissingDigests()` as well.

Reply:
I did as you suggested, but I do think shouldUploadLocalResultsToRemoteCache isn't named very well as it is being used to check inputs to remote actions not "Results".

Reply:
`shouldUploadLocalResultsToRemoteCache` was intended for checking local results as the code only executed when upload local results before this PR.

Now there are two cases to upload files:
    1. Upload as inputs for remote actions.
    2. Upload outputs of local actions.

IMO, the updated code is more clear for the two purposes. Thanks for the update!